components/nimbus/src/behavior.rs

```diff
@@ -273,338 +287,44 @@ impl EventStore {
-
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Months)
-                .unwrap()
-                .data
-                .buckets
-                .len(),
-            1
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Months)
-                .unwrap()
-                .data
-                .buckets[0],
-            1
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets
-                .len(),
-            3
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets[0],
-            1
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets[1],
-            0
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-1".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets[2],
-            0
-        ));
-
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-2".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets
-                .len(),
-            1
-        ));
-        assert!(matches!(
-            store
-                .events
-                .get(&"event-2".to_string())
-                .unwrap()
-                .intervals
-                .get(&Interval::Days)
-                .unwrap()
-                .data
-                .buckets[0],
-            0
-        ));
-
-        Ok(())
+    pub fn query(
+        &self,
+        event_id: String,
+        interval: Interval,
+        num_buckets: usize,
+        starting_bucket: usize,
+        query_type: EventQueryType,
+    ) -> Result<u64> {
+        if let Some(counter) = self.events.get(&event_id) {
+            if let Some(single_counter) = counter.intervals.get(&interval) {
```

Comment:
I'd consider having these in their respective classes:

```rs
impl MultiCounter {
    fn query(interval: Interval, num_buckets: usize, starting_bucket: usize, query_type: EventQueryType) -> Result<u64> {}
}

impl SingleCounter {
    fn query(num_buckets: usize, starting_bucket: usize, query_type: EventQueryType) -> Result<u64> {}
}
```

Reply:
So I did do that initially, I ended up pulling them all here because the event store/multi counter methods were sort of just wrappers for the actual logic in the single counter. If it makes more sense for the method to be there on each class I'm happy to move it back!

Reply:
I think the `EventQueryType` is the important one to grow a query function, so I'd be happy to lay off this refactor.